src/components/interaction-designer/toolbar/TreeBuilderToolbar.vue

```diff
@@ -235,39 +235,50 @@
                 <slot name="extra-dropdown-items-for-category2" />
               </div>
             </li>
-            <li
-              v-if="!isEmpty(blockClassesForBranchingCategory)"
-              class="nav-item dropdown">
-              <a
-                class="nav-link dropdown-toggle"
-                data-toggle="dropdown"
-                href="#"
-                role="button"
-                aria-haspopup="true"
-                aria-expanded="false"
-                @mouseover="openDropdownMenu($event.target)">
-                <span class="nav-link-text">{{ 'flow-builder.branching' | trans }}</span>
-              </a>
-              <div class="dropdown-menu mt-0">
-                <template v-for="(classDetails, className) in blockClassesForBranchingCategory">
-                  <div
-                    v-if="shouldDisplayDividerBefore(blockClassesForBranchingCategory, className)"
-                    :key="`${className}divider`"
-                    class="dropdown-divider" />
-                  <a
-                    v-if="isBlockAvailableByBlockClass[className]"
-                    :key="className + 'item'"
-                    href="#"
-                    class="dropdown-item"
-                    :data-block-type="className"
-                    :data-default-num-connections="classDetails['defaultConnections']"
-                    @click.prevent="handleAddBlockByTypeSelected(classDetails)">
-                    {{ translateTreeClassName(className) }}
-                  </a>
-                </template>
-                <slot name="extra-dropdown-items-for-category3" />
-              </div>
-            </li>
+            <template v-if="!isEmpty(blockClassesForBranchingCategory)">
+              <li class="nav-item" v-if="Object.keys(blockClassesForBranchingCategory).length === 1">
+                <a v-for="(classDetails, className) in blockClassesForBranchingCategory"
```

Comment:
even having only 01 element, I looped with v-for to get the content easily
Note: blockClassesForBranchingCategory is an associative array